src/Umbraco.Core/Migrations/Upgrade/V_8_7_0/StackedContentToBlockList.cs

```diff
@@ -73,16 +102,16 @@ private bool Migrate(IEnumerable<DataTypeDto> dataTypesToMigrate, Dictionary<Gui
             return refreshCache;
         }
 
-        private BlockListConfiguration UpdateConfiguration(DataTypeDto dataType, Dictionary<Guid, string> knownDocumentTypes)
+        private BlockListConfiguration UpdateConfiguration(DataTypeDto dataType, Dictionary<Guid, KnownContentType> knownDocumentTypes)
         {
             var old = JsonConvert.DeserializeObject<StackedContentConfiguration>(dataType.Configuration);
             var config = new BlockListConfiguration
             {
                 Blocks = old.ContentTypes?.Select(t => new BlockListConfiguration.BlockConfiguration
                 {
-                    Alias = knownDocumentTypes[t.IcContentTypeGuid],
+                    Alias = knownDocumentTypes.TryGetValue(t.IcContentTypeGuid, out var ct) ? ct.Alias : null,
                     Label = t.NameTemplate
-                }).ToArray(),
+                }).Where(c => c.Alias != null).ToArray(),
```

Comment:
null or whitespace?

Reply:
That would imply that there was a content type alias in the database that was whitespace, and which we didn't want to map here, because if the GUID matched a known content type it would get the alias in the DB, and if it didn't it would always return null.  I don't think whitespace aliases are valid for content types, but if they were, I'd want to allow them here.